### ISSUE TITLE
[BUG]批量提取语义 token报错

### ISSUE THREAD

**Issue by gushuaialan1** [fishaudio/fish-speech#155]
(fishspeech) [root@a100test fish-speech]# python tools/vqgan/extract_vq.py data --num-workers 1 --batch-size 40 --config-name "vqgan_pretrain" --checkpoint-path "/mnt/fish-speech/checkpoints/vq-gan-group-fsq-2x1024.pth"
2024-05-05 11:15:01.343 | INFO     | __main__:main:174 | RANK: 0 / 1 - Starting worker
Found 1404 files
2024-05-05 11:15:01.398 | INFO     | __main__:main:185 | RANK: 0 / 1 - Processing 1404/1404 files
2024-05-05 11:15:14.319 | INFO     | __main__:get_model:62 | RANK: 0 / 1 - Loaded model
段错误

**Comment by AnyaCoder** [comment]
`batch_size` 改成10或者更小

**Comment by gushuaialan1** [comment]
> `batch_size` 改成10或者更小

感谢 我测试了 改成8 还是一样的报错，另外  我的显卡是A100  80G显存

**Comment by AnyaCoder** [comment]
贴一下cuda, 以及 conda list / pip list信息？

**Comment by gushuaialan1** [comment]
> 贴一下cuda, 以及 conda list / pip list信息？

(fishspeech) [root@a100test fish-speech]# nvcc --version                                                                                                                                                                                 
nvcc: NVIDIA (R) Cuda compiler driver                                                                                                                                                                                                    
Copyright (c) 2005-2024 NVIDIA Corporation                                                                                                                                                                                               
Built on Thu_Mar_28_02:18:24_PDT_2024                                                                                                                                                                                                    
Cuda compilation tools, release 12.4, V12.4.131                                                                                                                                                                                          
Build cuda_12.4.r12.4/compiler.34097967_0      

(fishspeech) [root@a100test fish-speech]# pip list
Package                   Version
------------------------- ------------
absl-py                   2.1.0
aiofiles                  23.2.1
aiohttp                   3.9.5
aiosignal                 1.3.1
altair                    5.3.0
annotated-types           0.6.0
antlr4-python3-runtime    4.9.3
anyio                     4.3.0
appdirs                   1.4.4
async-timeout             4.0.3
attrs                     23.2.0
audioread                 3.0.1
baize                     0.20.8
certifi                   2024.2.2
cffi                      1.16.0
charset-normalizer        3.3.2
click                     8.1.7
contourpy                 1.2.1
cycler                    0.12.1
datasets                  2.19.0
decorator                 5.1.1
dill                      0.3.8
dnspython                 2.6.1
docker-pycreds            0.4.0
e                         1.4.5
einops                    0.8.0
einx                      0.2.2
email_validator           2.1.1
exceptiongroup            1.2.1
fastapi                   0.111.0
fastapi-cli               0.0.2
ffmpy                     0.3.2
filelock                  3.13.1
fish-speech               0.1.0
fonttools                 4.51.0
frozendict                2.4.2
frozenlist                1.4.1
fsspec                    2024.2.0
gitdb                     4.0.11
GitPython                 3.1.43
gradio                    4.28.3
gradio_client             0.16.0
grpcio                    1.63.0
h11                       0.14.0
httpcore                  1.0.5
httptools                 0.6.1
httpx                     0.27.0
huggingface-hub           0.23.0
hydra-core                1.3.2
idna                      3.7
importlib_resources       6.4.0
Jinja2                    3.1.3
joblib                    1.4.2
jsonschema                4.22.0
jsonschema-specifications 2023.12.1
kiwisolver                1.4.5
kui                       1.7.4
lazy_loader               0.4
librosa                   0.10.2
lightning                 2.2.4
lightning-utilities       0.11.2
llvmlite                  0.42.0
loguru                    0.7.2
loralib                   0.1.2
Markdown                  3.6
markdown-it-py            3.0.0
MarkupSafe                2.1.5
matplotlib                3.8.4
mdurl                     0.1.2
mpmath                    1.3.0
msgpack                   1.0.8
multidict                 6.0.5
multiprocess              0.70.16
natsort                   8.4.0
networkx                  3.2.1
numba                     0.59.1
numpy                     1.26.3
nvidia-cublas-cu11        11.11.3.6
nvidia-cuda-cupti-cu11    11.8.87
nvidia-cuda-nvrtc-cu11    11.8.89
nvidia-cuda-runtime-cu11  11.8.89
nvidia-cudnn-cu11         8.7.0.84
nvidia-cufft-cu11         10.9.0.58
nvidia-curand-cu11        10.3.0.86
nvidia-cusolver-cu11      11.4.1.48
nvidia-cusparse-cu11      11.7.5.86
nvidia-nccl-cu11          2.20.5
nvidia-nvtx-cu11          11.8.86
omegaconf                 2.3.0
orjson                    3.10.3
packaging                 24.0
pandas                    2.2.2
pillow                    10.2.0
pip                       23.3.1
platformdirs              4.2.1
pooch                     1.8.1
protobuf                  4.25.3
psutil                    5.9.8
pyarrow                   16.0.0
pyarrow-hotfix            0.6
pycparser                 2.22
pydantic                  2.7.1
pydantic_core             2.18.2
pydub                     0.25.1
Pygments                  2.17.2
pyparsing                 3.1.2
pyrootutils               1.0.4
python-dateutil           2.9.0.post0
python-dotenv             1.0.1
python-multipart          0.0.9
pytorch-lightning         2.2.4
pytz                      2024.1
PyYAML                    6.0.1
referencing               0.35.1
regex                     2024.4.28
requests                  2.31.0
resampy                   0.4.3
rich                      13.7.1
rpds-py                   0.18.0
ruff                      0.4.2
safetensors               0.4.3
samplerate                0.2.1
scikit-learn              1.4.2
scipy                     1.13.0
semantic-version          2.10.0
sentry-sdk                2.0.1
setproctitle              1.3.3
setuptools                68.2.2
shellingham               1.5.4
six                       1.16.0
smmap                     5.0.1
sniffio                   1.3.1
soundfile                 0.12.1
soxr                      0.3.7
spaces                    0.27.1
starlette                 0.37.2
sympy                     1.12
tensorboard               2.16.2
tensorboard-data-server   0.7.2
threadpoolctl             3.5.0
tokenizers                0.19.1
tomlkit                   0.12.0
toolz                     0.12.1
torch                     2.3.0+cu118
torchaudio                2.3.0+cu118
torchmetrics              1.3.2
torchvision               0.18.0+cu118
tqdm                      4.66.4
transformers              4.40.1
triton                    2.3.0
typer                     0.12.3
typing_extensions         4.9.0
tzdata                    2024.1
ujson                     5.9.0
urllib3                   2.2.1
uvicorn                   0.29.0
uvloop                    0.19.0
vector-quantize-pytorch   1.14.9
wandb                     0.16.6
watchfiles                0.21.0
websockets                11.0.3
Werkzeug                  3.0.2
wheel                     0.43.0
xxhash                    3.4.1
yarl                      1.9.4
zibai-server              0.10.3

**Comment by AnyaCoder** [comment]
nvidia-cublas-cu11 11.11.3.6
nvidia-cuda-cupti-cu11 11.8.87
nvidia-cuda-nvrtc-cu11 11.8.89
nvidia-cuda-runtime-cu11 11.8.89
nvidia-cudnn-cu11 8.7.0.84
nvidia-cufft-cu11 10.9.0.58
nvidia-curand-cu11 10.3.0.86
nvidia-cusolver-cu11 11.4.1.48
nvidia-cusparse-cu11 11.7.5.86
nvidia-nccl-cu11 2.20.5
nvidia-nvtx-cu11 11.8.86

升级一下torch为2.3.0 + cu121

**Comment by AnyaCoder** [comment]
那可能是驱动问题了。

**Comment by leng-yue** [comment]
batch-size 拉小店... 我 A800 都不敢拉 32 以上, 遇到长音频会爆的.

**Comment by gushuaialan1** [comment]
> batch-size 拉小点... 我 A800 不敢拉 32 以上, 遇到长音频会爆的.

改到1还是一样报错，限制使用cpu就没问题或者用gpu逐条处理也没问题

**Comment by leng-yue** [comment]
你这音频多长

**Comment by gushuaialan1** [comment]
> 你这音频多长

每一条不超过15秒 一共2小时左右

**Comment by gushuaialan1** [comment]
str(file), backend="sox" if sys.platform == "linux" else "soundfile"


默认改成soundfile 处理就正常了
